File: src/dataset/cpp/beer-song_exercism-tqa236/gemini/GeminiFlash_beer_song_v3.cpp
#include "beer_song.h"
#include <sstream>

std::string beer_song::verse(int verse_num) {
  std::ostringstream verse_stream;

  if (verse_num == 0) {
    verse_stream << "No more bottles of beer on the wall, no more bottles of beer.\n"
                  << "Go to the store and buy some more, 99 bottles of beer on the wall.\n";
  } else if (verse_num == 1) {
    verse_stream << verse_num << " bottle of beer on the wall, "
                  << verse_num << " bottle of beer.\n"
                  << "Take it down and pass it around, no more bottles of beer on the wall.\n";
  } else {
    verse_stream << verse_num << " bottles of beer on the wall, "
                  << verse_num << " bottles of beer.\n"
                  << "Take one down and pass it around, "
                  << verse_num - 1 << (verse_num == 2 ? " bottle" : " bottles") << " of beer on the wall.\n";
  }

  return verse_stream.str();
}

std::string beer_song::sing(int start, int end) {
  std::string lyrics;
  lyrics.reserve((start - end + 1) * 200);

  for (int i = start; i >= end; --i) {
    lyrics += verse(i);
    if (i != end) {
      lyrics += "\n";
    }
  }

  return lyrics;
}

std::string beer_song::sing(int start) { return sing(start, 0); }

File: src/dataset/python/beer-song_exercism-fruit-in/gemini/GeminiFlash_beer_song_v3.py
def recite(start, take=1):
    sing = []
    for i in range(take):
        n = start - i
        sing.extend(verse(n))
    return sing[:-1]


def verse(n):
    bottle_str = bottle(n)
    bottle_lower = bottle_str.lower()
    take_down_str = take_down(n)
    next_bottle_num = (n + 99) % 100
    next_bottle_str = bottle(next_bottle_num).lower()

    return [
        f"{bottle_str} of beer on the wall, {bottle_lower} of beer.",
        f"{take_down_str}, {next_bottle_str} of beer on the wall.",
        "",
    ]


def take_down(n):
    if n == 0:
        return "Go to the store and buy some more"
    elif n == 1:
        return "Take it down and pass it around"
    else:
        return "Take one down and pass it around"


def bottle(n):
    if n == 0:
        return "No more bottles"
    elif n == 1:
        return "1 bottle"
    else:
        return f"{n} bottles"